### PR TITLE
Issue 13725 content type flter

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/portlets/content-types/main/content-types.component.ts [hunk 1]
```diff
@@ -160,9 +162,9 @@ export class ContentTypesPortletComponent implements OnInit {
     }
 
     private setContentTypes(s: StructureTypeView[]): ButtonModel[] {
-        return s.map((structureTypeView) => {
+        return s.map(structureTypeView => {
```
Comment (alfredo-dotcms) [inline src/app/portlets/content-types/main/content-types.component.ts]: add type

File: src/app/portlets/content-types/main/content-types.component.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export class ContentTypesPortletComponent implements OnInit {
             this.dotContentletService.getAllContentTypes(),
             this.dotLicenseService.isEnterprise(),
             this.pushPublishService.getEnvironments().map((environments: DotEnvironment[]) => !!environments.length)
-        ).subscribe((res) => {
+        ).subscribe(res => {
```
Comment (alfredo-dotcms) [inline src/app/portlets/content-types/main/content-types.component.ts]: weird that your IDE removes all the parentesis

File: src/app/view/components/listing-data-table/listing-data-table.component.ts [hunk 3]
```diff
@@ -87,18 +88,30 @@ export class ListingDataTableComponent extends BaseComponent implements OnChange
             this.paginatorService.sortField = sortField;
             this.paginatorService.sortOrder = sortOrder === 1 ? OrderDirection.ASC : OrderDirection.DESC;
 
-            this.paginatorService.getWithOffset(offset).subscribe((items) => this.setItems(items));
+            this.paginatorService.getWithOffset(offset).subscribe(items => this.setItems(items));
         }
     }
 
+    /**
```
Comment (fmontes) [inline src/app/view/components/listing-data-table/listing-data-table.component.ts]: This doc is not Document this format, i think.

File: src/app/portlets/content-types/main/content-types.component.ts [hunk 4]
```diff
@@ -105,8 +105,10 @@ export class ContentTypesPortletComponent implements OnInit {
     }
 
     public changeBaseTypeSelector(value: string) {
-        this.listing.paginatorService.setExtraParams('type', value);
-        this.listing.loadData(0);
+        value !== ''
+            ? this.listing.paginatorService.setExtraParams('type', value)
+            : this.listing.paginatorService.deleteExtraParams('type');
+        this.listing.renewUrlsAndLoadData();
```
Comment (fmontes) [inline src/app/portlets/content-types/main/content-types.component.ts]: This is more like : `loadFirstPage` right?

File: src/app/api/services/paginator/paginator.service.ts [hunk 5]
```diff
@@ -72,6 +72,16 @@ export class PaginatorService {
...
 
+    /**
+     * Delete extra parameters of the eventual request.
+     * @param {string} name
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    deleteExtraParams(name: string): void {
+        this.extraParams.delete(name);
+    }
```
Comment (freddyucv) [inline src/app/api/services/paginator/paginator.service.ts]: test?

File: src/app/view/components/listing-data-table/listing-data-table.component.ts [hunk 6]
```diff
@@ -87,18 +88,30 @@ export class ListingDataTableComponent extends BaseComponent implements OnChange
...
+     * @memberof ListingDataTableComponent
+     */
+    renewUrlsAndLoadData(): void {
+        this.loading = true;
+        this.paginatorService.get().subscribe(items => {
+            this.setItems(items);
+            this.dataTable.first = 1;
+        });
+    }
+
```
Comment (freddyucv) [inline src/app/view/components/listing-data-table/listing-data-table.component.ts]: test?

**Pull request merged**